src/LibraryManager/Configuration/Settings.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Xml;
```

Comment:
Why XML instead of JSON for settings storage? Seems weird to use XML here when libman itself uses JSON for library settings storage.

Reply:
Couple factors:
1) Consistent with many other tools.  I modeled this mostly after the NuGet experience (admittedly, NuGet originally used xml files for packages.config, so maybe that's why they used it for nuget.config as well), but there's a lot of other tools that still use XML-based configuration.  Relatedly, because this is so much different from a libman.json file, it emphasizes that these files cannot be mixed up.
2) Easier APIs - honestly, this was simple to write because the XML APIs are so easy to work with.  I didn't want to do a straight serialization/deserialization with a settings store type.  This file should be extensible (possibly for future features or even for configuring/overriding provider-specific behaviors.  For example, if a user wants to change the cache lifetime for files, this is an appropriate place).
3) Honestly, I find XML easier to read.  JSON is more concise, but almost too sparse sometimes.

Reply:
Based on our offline conversation, I've re-implemented the data store using JSON instead of XML.  Take a look, if you like it, I want to clean up the history a bit before the final merge.